openedx/core/djangoapps/credit/models.py

```diff
@@ -65,28 +67,35 @@ class CreditProvider(TimeStampedModel):
 class CreditRequirement(TimeStampedModel):
     """This model represents a credit requirement.
 
-    Each requirement is uniquely identified by a `namespace` and a `name`. CreditRequirements
-    also include a `criteria` dictionary, the format of which varies by the type of requirement.
-    The criteria dictionary provides additional information clients may need to determine
-    whether a user has satisfied the requirement.
+    Each requirement is uniquely identified by its 'namespace' and
+    'location'.
+    The 'location' field stores the location of XBlock as a requirement, which
+    serves as the unique identifier for a credit course requirement.
+    The 'name' field stores the display name of the requirement.
+    The 'criteria' field dictionary provides additional information, clients
+    may need to determine whether a user has satisfied the requirement.
     """
 
     course = models.ForeignKey(CreditCourse, related_name="credit_requirements")
     namespace = models.CharField(max_length=255)
     name = models.CharField(max_length=255)
+    location = models.CharField(max_length=255)
```

Comment:
I think this is just a naming issue, but I'm concerned about calling this "location".  Not every credit eligibility requirement corresponds to a location in a course (the final grade is one example).

Instead, I'd prefer to:
1) Continue using (namespace, name) to uniquely identify the requirement.
2) For in-course reverification, continue assigning the XBlock location as the name of the requirement.
3) Add a "display_name" field to the requirement.  For ICRV, we will set this to the related assessment name.

Please let me know if you want to discuss this more.